### ISSUE TITLE
tearDown is not executed on interrupt

### ISSUE THREAD

**Issue by Syphdias** [kward/shunit2#108]
I expected `tearDown` and `oneTimetearDown` always to be run, even if the script was interrupted, e.g. by a SIGINT (pressing ctrl+c). Unfortunately this is not the case. Am I doing something wrong?

I also did not find any documentation if this is the responsibility of the individual script to write own traps.
I tried to to write simple traps but they seemingly were never triggered.

zsh: zsh 5.6.2 (x86_64-debian-linux-gnu)
shunit2: c1da1e0aa5eb1a48417261e53c55a0e00014105e

Example what I tested with:
```zsh
#!/usr/bin/env zsh
setopt shwordsplit
SHUNIT_PARENT=$0

function setUp() {
    echo this is setUp
}

function tearDown() {
    echo this is tearDown
}

trap 'echo caught INT' INT
TRAPINT() {
  echo caught INT
  return $((128 + $1))
}

testSleeping() {
  sleep 5
  echo sleeping done
  assertEquals 1 1
}

source shunit2/shunit2
```

**Comment by kward** [comment]
The traps, esp. Ctrl+C, should be caught as cancelling a suite of tests is very much a typical expectation.
https://github.com/kward/shunit2/blob/master/shunit2#L1259

I'll see what I can reproduce.

**Comment by kward** [comment]
It turned out that `tearDown()` and `oneTimeTearDown()` were never called.

**Comment by kward** [comment]
Argh. I submitted the code before running the other tests, and of course some are now failing.

**Comment by kward** [comment]
Fixed in de1fa04.
